### PR TITLE
Solving 03-components (Removable-emails, UiCounter & Weather-components) 

### PR DESCRIPTION
**Pull request by DariaaaP**
1) Added functionality to delete emails by clicking the button using id;
2) Added functionality for component-counter;
3) Divided the app into components (at my discretion)

### CODE REVIEW

**Comment by jsru-1** [comment]
Добавляю преподавателя (@ShGKme) для код-ревью.

File: 03-components/50-weather-components/WeatherApp.js [hunk 1]
```diff
@@ -1,52 +1,30 @@
...
   name: 'WeatherApp',
 
+  components: {
+    WeatherList,
+  },
+
+  setup() {
+    const cities = getWeatherData()
+
+    const weatherIcons = WeatherConditionIcons
```
Comment (ShGKme) [inline 03-components/50-weather-components/WeatherApp.js:16]: These icons are good in the components where they are actually used. It's unlikely that a user will use these component with different icons set. Especially if a user **have** to provide these icons, and there is no default value.

And even more if it's a part of interface of all the intermediate components like `WeatherList`.

File: 03-components/50-weather-components/components/WeatherCard.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+              <div class="weather-card__time">
+                {{ city.current.dt }}
+              </div>
+            </div>
+            <div class="weather-conditions">
+              <div class="weather-conditions__icon" :title="city.current.weather.description">{{ weatherIcons[city.current.weather.id] }}</div>
+              <div class="weather-conditions__temp">{{ cels }} °C</div>
+            </div>
+            <UiWeatherDetails :city="city.current"/>
+  `,
```
Comment (ShGKme) [inline 03-components/50-weather-components/components/WeatherCard.js:53]: Although it works, it's preferred to add a single root element for a component. Then a component can be used as an element, receive classes and attributes.

**Review by ShGKme — APPROVED** [review]
Approved, but see comments

**Pull request merged**